File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& inter, vector<int>& newi) {
        if(inter.empty()) return {newi};
        inter.push_back(newi);
        sort(inter.begin(),inter.end(),[](const vector<int> &i1, const vector<int> &i2){
            return i1[0]<i2[0];
        });
        vector<vector<int>> list = { inter[0] };
        for (size_t i = 1; i < inter.size(); i++) {
            if (inter[i][0] <= list.back()[1]) {
                list.back()[1] = max(inter[i][1], list.back()[1]);
            }else list.push_back(inter[i]);
        }
        return list;
    }
};

File: 0057-insert-interval/0057-insert-interval.js
/**
 * @param {number[][]} intervals
 * @param {number[]} newInterval
 * @return {number[][]}
 */
var insert = function(inter, newInterval) {
    inter.push(newInterval)
    inter.sort(function(a, b){return a[0] - b[0]})
    let list=[inter[0]]
    for(let i=1;i<inter.length;i++){
        if(inter[i][0]<=list[list.length-1][1]) list[list.length-1][1]=Math.max(inter[i][1],list[list.length-1][1])
        else list.push(inter[i])
    }
    return list
};